src/App.tsx

```diff
@@ -1,10 +1,57 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { Footer } from './components/Footer';
+import { TodoList } from './components/TodoList';
+import { getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoFilter } from './types/Filters';
+import { filterTodos } from './helpers/filterTodos';
+import { ErrorNotification } from './components/ErrorNotification';
+import { Header } from './components/Header';
 
-const USER_ID = 0;
+const USER_ID = 10926;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState<TodoFilter>(TodoFilter.All);
+  const [error, setError] = useState<string | null>(null);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => setError('Unable to load todos'));
+  }, []);
+
+  useEffect(() => {
+    let timerId: NodeJS.Timeout;
+
+    if (error) {
+      timerId = setTimeout(() => setError(null), 3000);
+    }
+
+    return () => clearTimeout(timerId);
+  }, [error]);
+
+  const visibleTodos = useMemo(() => (
+    filter === TodoFilter.All
+      ? todos
+      : filterTodos(todos, filter)
+  ), [todos, filter]);
```

Comment:
Different abstraction levels, why can't we handle the `TodoFilter.All` case in `filterTodos`?